src/CosmosDbStorage.cs

```diff
@@ -239,6 +242,20 @@ public static CosmosDbStorage Create(CosmosClient cosmosClient, string databaseN
 
     private async Task InitializeAsync(CancellationToken cancellationToken = default)
     {
+        if (!StorageOptions.CreateIfNotExists)
+        {
+            // check if container exists within database
+            try
+            {
+                Container = await Client.GetContainer(databaseName, containerName).ReadContainerAsync(cancellationToken: cancellationToken);
+                return;
```

Comment:
Cannot exit from here, as there are some dependencies that need to be created on the container.

Reply:
The idea behind this change is that I have a hangfire server app (responsible for job processing) and a hangfire client app (job scheduling only).

As for a hangfire server, I want to use `CreateIfNotExists=true` to preserve my database up-to-date with possible changes on this library side.

On the other hand, my clients are just clients, and I want to make them:
1) use ManagedIdentity (now it supports only data-plane operations)
2) have a fast startup/first query time (no need to upload multiple stored procedures and so on)

There is a DistributedCache library where the same flag is implemented: https://github.com/Azure/Microsoft.Extensions.Caching.Cosmos/blob/master/src/CosmosCache.cs#L447
Of course, there are no stored procedures, but the database might also be broken due to this flag (incorrect pk, etc.)

Reply:
Anyway, another option (suggestion) is to store some kind of hash, that is based on a procedure's content as a single entry in a database (something similar to EF migration tables). So, if the hash entry is missing or the hash value differs from the calculated value then there is no need to upload stored procedures. It will use a bit more CPU time and 1 GET request instead of calling Get + Create/Replace for every procedure.

Using this hash-based approach we could get rid of several non-friendly ManagedIdentity calls and speedup initialization in general.